extensions/amp-accordion/1.0/accordion.js

```diff
@@ -397,16 +415,47 @@ export function AccordionContent({
 }) {
   const ref = useRef(null);
   const hasMountedRef = useRef(false);
-  const {contentId, headerId, expanded, animate, setContentId} = useContext(
-    SectionContext
-  );
+  const {
+    contentId,
+    headerId,
+    expanded,
+    animate,
+    setContentId,
+    expandHandler,
+    experimentDisplayLocking,
+  } = useContext(SectionContext);
   const classes = useStyles();
+  const [contentVisibility, setContentVisibility] = useState(false);
+  const useDisplayLocking = contentVisibility && experimentDisplayLocking;
 
   useEffect(() => {
     hasMountedRef.current = true;
     return () => (hasMountedRef.current = false);
   }, []);
 
+  useEffect(() => {
+    if (!ref.current) {
+      return;
+    }
+    const element = ref.current;
+    const win = element.ownerDocument.defaultView;
+    const supportsContentVisibility = win.CSS.supports(
+      'content-visibility',
+      'hidden-matchable'
+    );
+    setContentVisibility(supportsContentVisibility);
+    if (!experimentDisplayLocking || !supportsContentVisibility) {
+      return;
+    }
+    const beforeMatchHandler = () => {
+      if (element.hasAttribute('hidden')) {
```

Comment:
Why is `hidden` important to check here? Why don't we want to expand the section in all cases, where an already expanded content element is just noop? Can we i.e. reuse `expand` or give `expandHandler` an `opt_forceExpand`?

Also, I assumed the `content-visibility: hidden-matchable` content is inside a `[hidden]` element, it shouldn't be findable. I manually checked this and it looks like the `display: block !important` overrides this. Interesting.

Reply:
Regarding checking the `hidden`, the `expandHandler` is actually a toggler.  And it's a toggler because it is re-used from the `header`'s onClick so it has to handle both expand and collapse action.

Yes, the we could give it some type of `opt-forceExpand`, although in this case, I think it's easier to check whether the `hidden` attribute is there or not.  The primary reason I prefer this is because it keeps all of the logic for handling the expansion based on display locking in one place, instead of moving part of it up to the definition of the `expandHandler` (which is in the Section).  And also the `hidden` attribute is a reliable indicator.  Another reason is that the `expandHandler` does not have any dependencies on the current `expanded` state of the current section.  So adding in the `opt-forceExpand` into this function would require the function to also track the `expanded` state, which increases complexity of this function.  In my opinion, it's cleaner to check for `hidden` then to update the logic of `expandHandler` with potentially messier code.

Regarding the `display: block !important` very true, very interesting!

Reply:
With display locking we can no longer use the `hidden` attribute.

Reply:
Removed `hidden` attribute in latest update.  Checking for `expanded` attribute on parent in this part now.